mbuild/compound.py

```diff
@@ -1009,6 +1010,77 @@ def generate_bonds(self, name_a, name_b, dmin, dmax):
                     self.add_bond((p1, p2))
                     added_bonds.append(bond_tuple)
 
+    @experimental_feature()
+    def freud_generate_bonds(
+        self,
+        name_a,
+        name_b,
+        dmin,
+        dmax,
+        exclude_ii=True,
+    ):
+        """Add Bonds between all pairs of types a/b within [dmin, dmax].
+
+        Parameters
+        ----------
+        name_a : str
+            The name of one of the Particles to be in each bond
+        name_b : str
+            The name of the other Particle to be in each bond
+        dmin : float
+            The minimum distance between Particles for considering a bond
+        dmax : float
+            The maximum distance between Particles for considering a bond
+        exclude_ii : bool, optional, default=True
+            Whether or not to include neighbors with the same index.
+        """
+        freud = import_("freud")
+        if self.box is None:
+            box = self.get_boundingbox()
+        else:
+            box = self.box
+        moved_positions = self.xyz - np.array(
+            [box.Lx / 2, box.Ly / 2, box.Lz / 2]
+        )
+
+        # extend non-periodic dimensions for pseudo-periodicity
```

Comment:
You might try setting the freud Box object's `periodic` flags. I haven't tested that use case very extensively, but it might work. If it doesn't work as expected, we should update one of the GitHub issues about aperiodic box support in freud.

Reply:
I thought i had issues with this previously, but it appears to work as expected now. Most likely user error 😜